src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,39 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.ValidationFailedException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DEFAULT_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new ValidationFailedException("User is null");
+        }
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationFailedException("This login is already taken: " + user.getLogin());
+        }
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new ValidationFailedException("Login and password can't be null");
```

Comment:
Don't create one big if with all conditions. Better to split one big if into several small ones. Each condition should throw a different exception with an informative message.